### PR TITLE
Chore: (deps): Bump k8s.io/client-go from 0.25.3 to 0.26.1

### PR DESCRIPTION
**Pull request by fengshunli**

### CODE REVIEW

**Review by FogDong — CHANGES_REQUESTED** [review]
Please don't downgrade other packages like cue when you try to update the vendor.

**Comment by fengshunli** [comment]
done

File: go.mod [hunk 1]
```diff
@@ -4,125 +4,126 @@ go 1.19
...
-	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.33 // indirect
+	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.35 // indirect
 	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
-replace (
-	cloud.google.com/go => cloud.google.com/go v0.100.2
-	sigs.k8s.io/apiserver-network-proxy/konnectivity-client => sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.24
-)
```
Comment (wonderflow) [inline go.mod:128]: ping @Somefive please check if the replace still needs

File: multicluster/client.go [hunk 2]
```diff
@@ -99,11 +99,19 @@ func (m *gatedClient) RESTMapper() meta.RESTMapper {
 	return m.base.RESTMapper()
 }
 
-func (m *gatedStatusWriter) Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+func (m *gatedClient) SubResource(subResource string) client.SubResourceClient {
+	return m.base.SubResource(subResource)
```
Comment (Somefive) [inline multicluster/client.go:103]: The implementation here is incorrect.

**Review by Somefive — CHANGES_REQUESTED** [review]
The upgrade of k8s dependencies from 0.25 to 0.26 contains a list of issues. And the same with the upgrade of controller-runtime to 0.14.

File: multicluster/client.go (hunk 2 above)
Comment (fengshunli) [reply multicluster/client.go:103]: SubResource is extra, and did not call

Comment (chivalryq) [reply multicluster/client.go:103]: But still it's not a good idea to put a random implementation here.

Comment (fengshunli) [reply multicluster/client.go:103]: image
The source code is placed after the Status method

Comment (Somefive) [reply multicluster/client.go:103]: SubResource client is used for subresource requests so for the gated multicluster client, we also need a gated subresource multicluster client and here you should call the delegated client.

**Comment by Somefive** [comment]
Duplicated with #69
Close as #69 merged.

**Pull request closed without merging**